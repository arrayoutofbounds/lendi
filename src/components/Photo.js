import React, { Component } from 'react';
import '../styles/index.css';

class Photo extends Component {

  render() {
    return (

        <div className="col-md-4">
        <div className="thumbnail">
          <a href={this.props.photo.thumbnailUrl} target="_blank">
            <img src={this.props.photo.thumbnailUrl} alt="" style={{width:"100%"}} />
            <div className="caption">
              <p><a href={this.props.photo.url}>{this.props.photo.title}</a></p>
            </div>
          </a>
        </div>
      </div>


    )
  }
}


export default Photo;