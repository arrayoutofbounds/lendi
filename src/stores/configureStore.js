
import { createStore, applyMiddleware } from 'redux';

import rootReducer from './photos/reducers';
import thunk from 'redux-thunk';

export default function configureStore(){
    return createStore(
        rootReducer, 
        applyMiddleware(thunk)
    );
}