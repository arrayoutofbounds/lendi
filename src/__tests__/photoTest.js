import React from 'react';
import Photo from '../components/Photo';
import expect from 'expect';
import configureStore from 'redux-mock-store'


import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

// const initialState = {}; 

// const mockStore = configureStore();
// let wrapper;
// let store;
// beforeEach(() => {
//   store = mockStore(initialState)
//   wrapper =  shallow( <Photo />);
//  })

const wrapper = shallow(<Photo />);
describe("<Photo/> Exists", () =>{
  it("renders 1 photo component", () => {
    // expect(wrapper.length).toEqual(1);
    console.log(wrapper.props())
    //expect(wrapper.props().data).toBe('something');
  });
});