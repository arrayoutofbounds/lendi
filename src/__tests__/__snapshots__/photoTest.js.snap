// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Photo/> Exists matches the snapshot 1`] = `
<div
  className="col-lg-3 col-md-4 col-xs-6 pitem"
>
  <a
    className="d-block mb-4 h-100"
    href="http://placehold.it/150/92c952"
    target="_blank"
  >
    <img
      alt="\${this.props.photo.title}"
      className="img-fluid img-thumbnail a1"
      src="http://placehold.it/150/92c952"
    />
  </a>
  <a
    className="a2"
    href="http://placehold.it/600/92c952"
    target="_blank"
  >
    <p
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      accusamus beatae ad facilis cum similique qui sunt
    </p>
  </a>
</div>
`;
